transcom/mymove
# INT B-21942

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21942)

## Summary

The powers that be want the Prime API to be able to see the destination GBLOC associated with shipment destination addresses.  The GBLOC will be based off of the `uspr_zip_id` on the from the `us_post_region_cities` table. It's important that every address have these values.

Exceptions:
- all USMC moves go to the USMC destination GBLOC
- all USAF/USSF in AK Zone II go to the MBFL GBLOC
- all other branches/zones go to their respective GBLOCs (lookin' at you `postal_code_to_gblocs`


This PR does the following:
- adds a db function that returns the GBLOC associated with addresses, pending they have a `us_post_region_cities_id` - separates logic for CONUS & OCONUS shipments and exceptions for AK Zone II with USAF/USSF
- updates addresses struct with a `DestinationGbloc` value - adding a ``db:"-"`` tells Pop to go kick rocks and it knows it's just a calculated/computed value
- updated `getMoveTaskOrder` service object to send back the destination GBLOC on the destination address
- tests

### How to test

1. Create a move, add a shipment or two with and without an AK Zone II address (query `re_oconus_rate_areas` with `WHERE rate_area_id = '5a27e806-21d4-4672-aa5e-29518f10c0aa'` to see all `us_post_region_cities.id` associated with AK Zone II
2. Approve as SC, approve as TOO to make the move available to Prime
3. Fire up [Prime V3 Swagger UI](http://primelocal:3000/swagger-ui/prime_v3.html#/moveTaskOrder/getMoveTaskOrder) or Postman
4. Query the move associated with the shipments you just created
5. You should now see the `destinationGbloc` value in `mtoShipments[i].destinationAddress` object in the return value

## Screenshots
![Screenshot 2024-12-20 at 4 38 10 PM](https://github.com/user-attachments/assets/41af6720-479e-4115-8a80-c620cf1a926d)
